src/Silverback.Integration/Messaging/Outbound/TransactionalOutbox/OutboxProduceStrategy.cs

```diff
@@ -46,7 +45,10 @@ public Task ProduceAsync(IOutboundEnvelope envelope)
 
                 _logger.LogWrittenToOutbox(envelope);
 
-                _producer ??= _outboundQueueBroker.GetProducer(envelope.Endpoint);
+                if (_producer == null || _producer.Endpoint.Name != envelope.Endpoint.Name)
```

Comment:
I wouldn't rely on the `Endpoint.Name` only but I think that we should compare the whole object (it should implement `IEquatable`, so you can just compare).

Reply:
Updated and amended tests